lib/wallet/src/Cardano/Wallet.hs

```diff
@@ -2253,12 +2253,14 @@ buildTransactionPure
   where
     nodeProtocolParameters =
         ( protocolParams
-        , fromMaybe
+        , maybe
             (error $ unwords
                 [ "buildAndSignTransactionPure: no nodePParams."
                 , "should only be possible in Byron, where"
                 , "withRecentEra should prevent this to be reached."
                 ])
+            (Cardano.bundleProtocolParams
+                (WriteTx.fromRecentEra (WriteTx.recentEra @era)))
```

Comment:
(This seems like the more relevant location for that comment https://github.com/input-output-hk/cardano-wallet/pull/3812/files#r1153379942 )

Reply:
Actually, for fee estimation's sake, simply I guess just moving this conversion out of the `calculateFeePercentiles` loop would help 🤔

Reply:
Don't see any effect here: https://github.com/input-output-hk/cardano-wallet/pull/3822